tests/test_movielens.py

```diff
@@ -704,7 +704,7 @@ def split(self, X, y=None, groups=None):
             for _ in range(self.n_splits):
                 yield idx, idx
 
-    cv = CV(n_splits=3, random_state=42)
+    cv = CV(n_splits=3, random_state=42, shuffle=True)
```

Comment:
We can't make this change as it would make tests non-deterministic.

Could you explain in more detail why this is needed? That will help us find another solution.

Reply:
I verified what's going on, and this will not be non-deterministic. Let's merge.